dll/win32/shell32/CMoveToMenu.cpp

```diff
@@ -33,6 +36,36 @@ WindowProc(HWND hwnd, UINT uMsg, WPARAM wParam, LPARAM lParam)
         case WM_ENABLEOK:
             SendMessageW(hwnd, BFFM_ENABLEOK, 0, (BOOL)lParam);
             return 0;
+
+        case WM_COMMAND:
+        {
+            switch (LOWORD(wParam))
+            {
+                case IDC_BROWSE_FOR_FOLDER_FOLDER_TEXT:
+                {
+                    if (HIWORD(wParam) == EN_CHANGE)
+                    {
+                        if (!this_->m_bIgnoreTextBoxChange)
+                        {
+                            // get the text
+                            GetDlgItemTextW(hwnd, IDC_BROWSE_FOR_FOLDER_FOLDER_TEXT, szPath, _countof(szPath));
+                            StrTrimW(szPath, L" \t");
+
+                            // update OK button
+                            BOOL bValid = !PathIsRelative(szPath) && PathIsDirectoryW(szPath);
+                            SendMessageW(hwnd, BFFM_ENABLEOK, 0, bValid);
```

Comment:
Instead of adding an additional path for this,
would it be an idea to convert it to a pidl, and set this pidl as if it was selected in the treeview? (and possibly even select it)
In that way, it might only need some code in the EN_CHANGE here, and the rest could magically work.

Reply:
After `BFFM_SELCHANGED` callback, `OK` button will be enabled or disabled by the system. `PostMessageW` will be overwrite the `OK` button enabled/disabled status by message queueing.

Reply:
We have to distinguish between the relative path and the absolute path. Setting the pidl will store the file title (not absolute path) into the text box. Setting the pidl will cause text box change, text box change will set the pidl, then an infinite recursive loop will happen.

Reply:
>would it be an idea to convert it to a pidl, and set this pidl as if it was selected in the treeview? (and possibly even select it)

It needs another flag to avoid the infinite loop...

Reply:
Setting a PIDL will store the file title. This behavior is inevitable.